### PR TITLE
feat(openapi): convert openapi to types and use in project

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Allow to use types/endpoints introduced with OpenAPI earlier

### 🖼️ Screenshots / Screencasts

image

### 🚧 Tasks

- [ ] Check other IDE (VSCode)
- [ ] Code review

### CODE REVIEW

File: src/components/ConversationSettings/BotsSettings.vue [hunk 1]
```diff
@@ -76,21 +76,19 @@ export default {
 	},
 
 	setup() {
-		const botsStore = useBotsStore()
-
 		return {
-			botsStore,
+			botsStore: useBotsStore(),
 		}
 	},
```
Comment (DorraJaouad) [inline src/components/ConversationSettings/BotsSettings.vue:82]: image

I think ts in SFC needs more refactoring, methods used in template are not inferred only when they are added to setup hook.

File: src/components/AdminSettings/BotsSettings.vue [hunk 2]
```diff
@@ -106,7 +108,14 @@ import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadi
 import NcPopover from '@nextcloud/vue/dist/Components/NcPopover.js'
 
 import { BOT } from '../../constants.js'
-import { getAllBots } from '../../services/botsService.js'
+import { getAllBots } from '../../services/botsService.ts'
+import type { BotWithDetails } from '../../types'
+
+type StateIcon = {
+	state_icon_component: ComponentOptions<Vue>,
```
Comment (DorraJaouad) [inline src/components/AdminSettings/BotsSettings.vue:115]: type conflict.

image

**Review by DorraJaouad — COMMENTED** [review]
`Option 'allowImportingTsExtensions' can only be used when either 'noEmit' or 'emitDeclarationOnly' is set.ts`

https://github.com/nextcloud/spreed/blob/fca0faeced796484720db266b970685c096f6a0d/tsconfig.json#L8

File: .eslintrc.js [hunk 3]
```diff
@@ -90,5 +90,14 @@ module.exports = {
 				'node/no-unpublished-import': 'off',
 			},
 		},
+		{
+			files: ['src/types/openapi/*.ts'],
+			rules: {
+				'@typescript-eslint/no-explicit-any': 'off',
+				quotes: 'off',
+				'no-multiple-empty-lines': 'off',
+				'no-use-before-define': 'off',
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply .eslintrc.js:99]: that's enabled only for auto-generated files

**Review by DorraJaouad — APPROVED** [review]
Looks good for now

File: tsconfig.json [hunk 4]
```diff
@@ -1,18 +1,13 @@
...
 		"allowJs": true,
 		"checkJs": true,
 		"allowImportingTsExtensions": true,
 		"lib": ["ESNext"],
-		"module": "ESNext",
-		"moduleResolution": "bundler",
-		"verbatimModuleSyntax": true,
-		"target": "ES2020",
-		"esModuleInterop": true,
-		"forceConsistentCasingInFileNames": true,
```
Comment (ShGKme) [inline tsconfig.json:15]: What was wrong here?

Comment (Antreesy) [reply tsconfig.json:15]: covered by default config: `"extends": "@vue/tsconfig/tsconfig.json"`

Also we rolback and disabled checking for .vue files later, as it keep complaining on Windows + VSCode. Please check from your side, if there is something we've done wrong, when you have time

File: src/components/VolumeIndicator/VolumeIndicator.vue
Comment (ShGKme) [inline src/components/VolumeIndicator/VolumeIndicator.vue:1]: Does this PR revert TS support for Vue?

Comment (Antreesy) [reply src/components/VolumeIndicator/VolumeIndicator.vue:1]: Right, that was exactly this PR :see_no_evil:

File: tsconfig.json (hunk 4 above)
Comment (ShGKme) [reply tsconfig.json:15]: This PR breaks building Talk Desktop and I cannot understand, how it does 👀

**Pull request merged**